diff --git a/extras/Build/juceaide/CMakeLists.txt b/extras/Build/juceaide/CMakeLists.txt
index 17e552097..feca9a224 100644
--- a/extras/Build/juceaide/CMakeLists.txt
+++ b/extras/Build/juceaide/CMakeLists.txt
@@ -63,7 +63,7 @@ if(JUCE_BUILD_HELPER_TOOLS)
 else()
     message(STATUS "Configuring juceaide")
 
-    if(CMAKE_CROSSCOMPILING)
+    if(CMAKE_CROSSCOMPILING AND NOT CMAKE_CROSSCOMPILING_EMULATOR)
         unset(ENV{ADDR2LINE})
         unset(ENV{AR})
         unset(ENV{ASM})
@@ -178,7 +178,7 @@ else()
 
     message(STATUS "Testing juceaide")
 
-    execute_process(COMMAND "${imported_location}" version
+    execute_process(COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} "${imported_location}" version
         RESULT_VARIABLE result_variable
         OUTPUT_VARIABLE output
         ERROR_VARIABLE output)
